src/api/endpoints/auth/TokenManager.ts

```diff
@@ -30,11 +32,12 @@ export default class TokenManager extends ApiEndpoint {
 
   public async getAccessToken(source: string = "unknown"): Promise<string | undefined> {
     this.log?.debug(`getAccessToken: New token requested from ${source}`);
-    if (this.accessToken !== undefined) {
-      const exp = TokenManager.getJwtFromAccessToken(this.accessToken).exp;
+    if (this.accessToken !== undefined && this.accessTokenExpiration !== undefined) {
+      const exp = Math.trunc(new Date(this.accessTokenExpiration).getTime() / 1000);
+      const now = Math.trunc(Date.now() / 1000);
 
       // check expiration minus 5 minutes to guard against race condition or timing issues creating unnecessary 403s
-      if (exp > Date.now() / 1000 - 60 * 5) {
+      if (exp > now - 300) {
```

Comment:
I'm pretty sure this is still broken, and we _actually_ want `exp - 300 > now`.
```suggestion
      if (exp - 300 > now) {
```

Talking through this to make sure I understand it:
`exp > now - 300` is saying "exp > 5 minutes ago" which I don't think is what we want
`exp - 300 > now` is saying `5 minutes until expiration > now` which I'm pretty sure is what we want

Reply:
Oh yeah definitely! That's what I get from working too late